### ISSUE TITLE
Email in LaTeX not working

### ISSUE THREAD

**Issue by phieri** [robinst/linkify#51]
When using lychee on a LaTeX book repository the following email addresses were not picked up correctly with the curly brackets.

https://github.com/SverigesSandareamatorer/SSA-Akademin/blob/32bef56e654e4be9b2a3e573a337a97f70d79650/koncept/foreword.tex#L156
Picks up `texttt{hq@ssa.se`

https://github.com/SverigesSandareamatorer/SSA-Akademin/blob/32bef56e654e4be9b2a3e573a337a97f70d79650/koncept/tryckort.tex#L58
Picks up `hq@ssa.se}{hq@ssa.se`

**Comment by mre** [comment]
Maintainer of lychee here. Not sure if linkify is the right place to solve this.
I think `texttt{hq@ssa.se` is technically a syntactically valid email address. The curly brace is not very common, but it's an [accepted special character](https://stackoverflow.com/a/2049510/270334).
Properly parsing LaTeX would avoid that problem, but I'm not aware of a good crate for that.

**Comment by robinst** [comment]
Yeah what @mre said. What's interesting though is that GitHub doesn't allow `{` and handles these cases as you'd expect:

texttt{hq@ssa.se

hq@ssa.se}{hq@ssa.se
